pkg/core/storage/boltdb_store.go

```diff
@@ -37,30 +33,50 @@ func NewBoltDBStore(cfg BoltDBOptions) (*BoltDBStore, error) {
 		return nil, err
 	}
 	err = db.Update(func(tx *bbolt.Tx) error {
-		_, err = tx.CreateBucketIfNotExists(Bucket)
-		if err != nil {
-			return fmt.Errorf("could not create root bucket: %w", err)
+		for i := 0; i <= 255; i++ {
+			_, err := tx.CreateBucketIfNotExists([]byte{byte(i)})
+			if err != nil {
+				return err
+			}
+			_, err = tx.CreateBucketIfNotExists([]byte{byte(DataMPT), byte(i)})
+			if err != nil {
+				return err
+			}
 		}
 		return nil
 	})
 
-	return &BoltDBStore{db: db}, nil
+	return &BoltDBStore{db: db}, err
+}
+
+func split(key []byte) ([]byte, []byte) {
+	switch KeyPrefix(key[0]) {
+	case DataMPT:
+		if len(key) >= 2 {
+			return key[:2], key
+		}
+		return key[:1], key
+	default:
+		return key[:1], key
```

Comment:
We have very limited number of cases where `key[1:]` is zero-length, maybe we should try fixing them and providing proper split here. I'm not sure it'll change anything, but who knows.